src/execution_plan/execution_plan.c

```diff
@@ -725,7 +814,8 @@ ExecutionPlan *NewExecutionPlan(RedisModuleCtx *ctx, GraphContext *gc, ResultSet
 	// Optimize the operations in the ExecutionPlan.
 	optimizePlan(gc, plan);
 
-	plan->segment_count = segment_count;
+	// Disregard self.
+	plan->segment_count = segment_count - 1;
```

Comment:
can you explain?

Reply:
the last computed segment is acting as the complete execution-plan, 
plan = segments[segment_count-1]
there's no need for plan's segment to contain a reference to itself.